webapp/src/components/App/Singlestudy/explore/Debug/index.tsx

```diff
@@ -12,122 +12,121 @@
+      reloadTree: treeDataResponse.reload,
+      isTreeLoading: treeDataResponse.isLoading,
     }),
-    [res.reload],
+    [treeDataResponse.reload, treeDataResponse.isLoading],
   );
 
+  // Update the selected file
   useUpdateEffect(() => {
-    const firstChildName = Object.keys(res.data ?? {})[0];
-    const firstChildTreeData = R.path<TreeData>([firstChildName], res.data);
+    // Select the file corresponding to the `path` URL parameter
 
     const pathSegments = path?.split("/");
     const filename = pathSegments ? R.last(pathSegments) : null;
-    const treeData = pathSegments ? R.path<TreeData>(pathSegments, res.data) : null;
-
-    let fileInfo: FileInfo | null = null;
+    const treeData = pathSegments ? R.path<TreeData>(pathSegments, treeDataResponse.data) : null;
 
     if (path && filename && treeData) {
-      fileInfo = {
+      setSelectedFile({
         fileType: getFileType(treeData),
         treeData,
         filename,
         filePath: path,
-      };
-    } else if (firstChildTreeData) {
-      fileInfo = {
+      });
+
+      return;
+    }
+
+    // Select the first child of the tree data
+
+    const firstChildName = Object.keys(treeDataResponse.data ?? {})[0];
+    const firstChildTreeData = R.path<TreeData>([firstChildName], treeDataResponse.data);
+
+    if (firstChildTreeData) {
+      setSelectedFile({
         fileType: getFileType(firstChildTreeData),
         treeData: firstChildTreeData,
         filename: firstChildName,
         filePath: firstChildName,
-      };
+      });
+
+      return;
     }
 
-    setSelectedFile(fileInfo);
-  }, [res.data, path]);
+    setSelectedFile(null);
+  }, [treeDataResponse.data, path]);
 
+  // Update the `path` URL parameter
   useUpdateEffect(() => {
     if (selectedFile?.filePath !== path) {
       setSearchParams({ path: selectedFile?.filePath || "" });
     }
   }, [selectedFile?.filePath]);
 
-  // TODO: to remove after ANT-2217 solved
+  // Solve a back issue when the archive status change (cf. ANT-2217)
   useUpdateEffect(() => {
-    contextValue.reloadTreeData();
+    contextValue.reloadTree();
   }, [study.archived]);
 
   ////////////////////////////////////////////////////////////////
   // JSX
   ////////////////////////////////////////////////////////////////
 
   return (
-    <SplitView id="debug" sizes={[20, 80]}>
-      <Box sx={{ minWidth: 150, p: 1, overflow: "auto" }}>
-        <UsePromiseCond
-          response={res}
-          ifFulfilled={(data) => (
-            <DebugContext.Provider value={contextValue}>
+    <DebugContext.Provider value={contextValue}>
+      <SplitView id="debug" sizes={[20, 80]}>
+        <Box sx={{ minWidth: 150, p: 1, overflow: "auto", position: "relative" }}>
+          <UsePromiseCond
+            keepLastResolvedOnReload
+            response={treeDataResponse}
+            ifPending={() =>
+              Array.from({ length: 3 }).map((_, index) => <Skeleton key={index} height={32} />)
```

Comment:
Even if its just a loader I dont know about using the index of an array as key property

Reply:
No risk for this

Reply:
ok